### DISCUSSION TITLE
How to checkpoint submodules

### DISCUSSION THREAD

**Discussion by dtch1997** [Lightning-AI/pytorch-lightning#7768] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
By default pytorch lightning Trainer saves checkpoint for the full model. However I would like to checkpoint manually for individual submodules. 

Eg. I might have an image classifier that predicts labels and a image caption generator that generates captions from images; both of them use the same CNN encoder. I might want to do something like pre-train with the classification task (saving checkpoints) and then load only the trained encoder (minus the prediction head) for use in the downstream caption task. Furthermore I might have several backbones that I want to test out for the encoder/ 

What's the idiomatic way to do this in PyTorch Lightning?

**Comment by justusschock** [comment]
Hi,

to do this, you probably have to provide your own [ModelCheckpoint Callback](https://pytorch-lightning.readthedocs.io/en/latest/extensions/generated/pytorch_lightning.callbacks.ModelCheckpoint.html#pytorch_lightning.callbacks.ModelCheckpoint)

**Reply by dtch1997** [reply]
thanks @justusschock . Are there code snippets of custom checkpoint callbacks that I can refer to?

**Reply by justusschock** [reply]
None that I know of, to be honest.

**Reply by dtch1997** [reply]
Alright thanks. I think I'll stick to using `torch.save` and `torch.load` for now then.
